compiler/lib/src/main/scala/codegen/CppWriter/ComputeCppFiles.scala

```diff
@@ -44,7 +44,13 @@ object ComputeCppFiles extends AstStateVisitor {
   ) = {
     val node = aNode._2
     val data = node.data
-    visitList(s, data.members, matchComponentMember)
+    val name = s.getName(Symbol.Component(aNode))
+    val loc = Locations.get(node.id)
+    for {
+      s1 <- addMappings(s, FileNames.getComponent(name), Some(loc))
```

Comment:
This looks good. Or you can replace s1 and s2 with s. If you reuse s in a for...yield construct, it creates a new s that hides the old s. This is a useful pattern for "assigning" to s. It avoids using the wrong version of s1, s2, etc.

Reply:
Ok, I changed this. I think that's better.